src/tools/annotation/CircleRoiTool.js

```diff
@@ -507,6 +510,17 @@ function _calculateStats(image, element, handles, modality, pixelSpacing) {
       ((pixelSpacing && pixelSpacing.colPixelSpacing) || 1)) /
     2;
   const perimeter = 2 * Math.PI * radius;
+
+  // Pixel diagonal
+  const pixelDiagonal = pixelSpacing.colPixelSpacing
+    ? decimal(
+        pixelSpacing.colPixelSpacing * pixelSpacing.colPixelSpacing +
+          pixelSpacing.rowPixelSpacing * pixelSpacing.rowPixelSpacing
+      ).sqrt()
+    : decimal.sqrt(2);
+
+  const uncertainty = perimeter * pixelDiagonal;
```

Comment:
all this logic could be extracted to a function, i.e. calculateUncertainty(pixelSpacing)
This would make the code easier to read and test. The comment "Pixel diagonal" seems redundant to me

Reply:
Was thinking the same actually.. and thanks for the encouragement! :D